modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/AbstractProjectEventSyncCommand.java

```diff
@@ -0,0 +1,67 @@
+package com.enonic.xp.core.impl.content;
+
+import com.google.common.base.Preconditions;
+
+import com.enonic.xp.content.ContentName;
+import com.enonic.xp.content.ContentPath;
+import com.enonic.xp.content.ContentService;
+
+public abstract class AbstractProjectEventSyncCommand
+{
+    final ContentService contentService;
+
+    final ProjectEventSyncCommandParams params;
+
+    AbstractProjectEventSyncCommand( final Builder builder )
+    {
+        this.contentService = builder.contentService;
+        this.params = builder.params;
+    }
+
+    public void sync()
+    {
+        this.params.getTargetContext().runWith( this::doSync );
+    }
+
+    protected abstract void doSync();
+
+    protected ContentPath buildNewPath( final ContentPath parentPath, final ContentName name )
+    {
+        String newName = name.toString();
+
+        while ( contentService.contentExists( ContentPath.from( parentPath, newName ) ) )
+        {
+            newName = NameValueResolver.name( newName );
+        }
+
+        return ContentPath.from( parentPath, newName );
+    }
+
+    public static abstract class Builder<B extends Builder<B>>
```

Comment:
Codacy found an issue: ['abstract' modifier out of order with the JLS suggestions.](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=6324410)